packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceCreateManifest.test.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { ComponentSet } from '@salesforce/source-deploy-retrieve';
+import { fail } from 'assert';
+import { expect } from 'chai';
+import * as fs from 'fs';
+import util = require('util');
+import * as sinon from 'sinon';
+import { createSandbox } from 'sinon';
+import * as vscode from 'vscode';
+import {
+  forceCreateManifest
+} from '../../../src/commands';
+import { nls } from '../../../src/messages';
+
+const classPath = '/force-app/main/default/classes/';
+const class1 = 'Apex1';
+const class2 = 'Apex2';
+const uri1 = vscode.Uri.parse(classPath + class1);
+const uri2 = vscode.Uri.parse(classPath + class2);
+const manifest = `<?xml version="1.0" encoding="UTF-8"?>
+<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+  <types>
+    %s
+    <name>ApexClass</name>
+  </types>
+</Package>`;
+
+describe('Force Create Manifest', () => {
+  describe('Happy Paths', () => {
+
+    let env: sinon.SinonSandbox;
+    let openDocSpy: sinon.SinonSpy;
+    let inputBoxSpy: sinon.SinonStub;
+
+    before(() => {
+      env = createSandbox();
+    });
+
+    beforeEach(() => {
+      inputBoxSpy = env.stub(vscode.window, 'showInputBox');
+      openDocSpy = env.spy(vscode.workspace, 'openTextDocument');
+    });
+
+    afterEach(() => {
+      env.restore();
+    });
+
+    it('Should create first manifest from single sourceUri', async () => {
+      const packageXML = util.format(manifest, `<member>${class1}</member>`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync').returns(false);
+      env.stub(fs, 'mkdirSync').returns(undefined);
+      env.stub(fs, 'writeFileSync').returns(undefined);
+      const packageName = 'package' + generateRandomSuffix() + '.xml';
+      inputBoxSpy.onCall(0).returns(packageName);
+      await forceCreateManifest(uri1, undefined);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
```

Comment:
should we verify that the other stubs were called here with the right parameters? If we open the wrong file in the wrong place that'd be sadness.

Reply:
Which stubs are you thinking are missing here? The one that I can't seem to add with the stubs in place is showing the text document, since I don't actually have anything to open without the file being created.